packages/base/lib/plugin.js

```diff
@@ -1,17 +1,21 @@
-import { logger } from '@appium/support';
-
+import {logger} from '@appium/support';
 
+/**
+ * @abstract
+ */
 export default class BasePlugin {
-
-  // plugins can define new methods for the Appium server to map to command names, of the same
-  // format as used in Appium's routes.js, for example, this would be a valid newMethodMap:
-  //
-  // {
-  //   '/session/:sessionId/new_method': {
-  //      GET: {command: 'getNewThing'},
-  //      POST: {command: 'setNewThing', payloadParams: {required: ['someParam']}}
-  //   }
-  // }
+  /**
+   * Plugins can define new methods for the Appium server to map to command names, of the same
+   * format as used in Appium's routes.js, for example, this would be a valid newMethodMap:
+   * @type {Record<string,import('express').IRoute}
```

Comment:
I don't think this type is correct. This is the idiosyncratic format used by Appium in our routes.js file, I think we have a type exported for that now we could use?